sklearn/svm/src/libsvm/svm.cpp

```diff
@@ -2260,7 +2260,7 @@ static void svm_group_classes(const PREFIX(problem) *prob, int *nr_class_ret, in
                 i = j-1;
                 this_label = label[j];
                 this_count = count[j];
-                while(i>=0 && label[i] > this_label)
+                while(i>=0 && label[i] < this_label)
```

Comment:
The reason for that is that y=0 is assigned to the class +1, and y=1 — to the class -1. As a result, sign of the decision function gets flipped.